syntax = "proto3";

package tutorial;

import "google/protobuf/empty.proto";
import "base.proto";

message NewsRequest {
  string data = 1;  // 订阅的新闻类型
}

message NewsResponse {
  string data = 1;  // 新闻数据
}

service NewsStore {
  rpc Ping(google.protobuf.Empty) returns (Pong);                         // Ping
  rpc GetNewsStream(NewsRequest) returns(stream NewsResponse);            // 服务端流模式
  rpc PutNewsStream(stream NewsRequest) returns(NewsResponse);            // 客户端流模式
  rpc GetPutNewsStream(stream NewsRequest) returns(stream NewsResponse);  // 双向流模式
}